Lidarlite pwm input functionnality

mgirlando opened the pull request:
I wrote a modification to add the Lidar Lite V2 (not tested on V1 but should work) on pwm input of the sonar.
You just plug the device like this :
![lidar_lite_pwm_wiring2](https://cloud.githubusercontent.com/assets/7744477/9567595/766bef26-4f31-11e5-9dca-04fcedff0d70.png)

The sofware automaticaly detects that the sensor is Lidar and not HCSR04.
No need to add features.
Maybe the name of driver (sonar_hcsr04.*) should be renamed to reflect that.

mgirlando commented:
you need to add those stub in  altitude_hold_unittest.cc to pass the test:

```
int32_t sonarGetMaxDistance(void)
{return 400;}

int32_t sonarGetTransitionDistance(void);
{return 300;}
```

thenickdude commented on src/main/drivers/sonar_hcsr04.c:
```diff
@@ -36,15 +35,25 @@
 static uint32_t lastMeasurementAt;
 static volatile int32_t measurement = -1;
 static sonarHardware_t const *sonarHardware;
+static bool SonarIsLidarLite = false;
```
Variable name should start with a lowercase letter.

thenickdude commented on src/main/drivers/sonar_hcsr04.c:
```diff
@@ -52,6 +61,27 @@ static void ECHO_EXTI_IRQHandler(void)
+            // are always surrounded by two good values)
+            // we detect them and remove the suspect ones
+
+            if(SonarIsLidarLite == true) {
```
should just be `if (sonarIsLidarLite) {`

thenickdude commented on src/main/drivers/sonar_hcsr04.c:
```diff
@@ -128,12 +158,24 @@ void hcsr04_init(const sonarHardware_t *initialSonarHardware)
 // measurement reading is done asynchronously, using interrupt
 void hcsr04_start_reading(void)
 {
+    //if(SonarIsLidarLite == true) return; // not used for Lidar
```
Line should be uncommented or removed entirely.

thenickdude commented on src/main/drivers/sonar_hcsr04.c:
```diff
@@ -156,17 +199,46 @@ void hcsr04_start_reading(void)
-    int32_t distance = measurement / 59;
+    int32_t distance;
+
+    if(SonarIsLidarLite == false) {
```
Should be `if (!sonarIsLidarLite) {`

thenickdude commented:
There seems to be very little overlap between the code you're currently combining for the Sonar and Lidar drivers, they should probably just be separated out into separate drivers.

thenickdude commented on src/main/sensors/sonar.c:
```diff
@@ -142,4 +142,19 @@ int32_t sonarGetLatestAltitude(void)
 }
 
+/**
+ * returns the maximum distance readable by the sonar
```
Would be nice to note units of the return type here (I believe it's centimetres).

mgirlando commented:
Made all the corrections you requested (coding rules)
Renamed sonar_hcsr04 to pwm_rangers  --> splitting it in 2 drivers would be more complicated as they share same interrupt. (I am creating one to use I2C input to free softserial)
Modified also the unit test (altitude_hold_unittest.cc) but I cannot compile the tests ( hope it works)
